### DISCUSSION TITLE
pgsc_calc execution time

### DISCUSSION THREAD

**Discussion by Than0sss** [PGScatalog/pgsc_calc#212] [Q&A] [answered]
Hi everyone,
How much computation time does it take on average to compute the PRS per genome and for one score in a linux machine with
16GB RAM
2 CPUs
Do you monitored the computation time  on the 1000 genomes dataset, in such a machine
Best

**Comment by smlmbrt** [comment, marked as answer]
Dear @Than0sss, we previously evaluated how `pgsc_calc v1.3.2` scaled per score in UK Biobank (480,000 samples, https://x.com/PGSCatalog/status/1597260221508091906?s=20) but we've made some improvements to bottlenecks since (patches forthcoming). The pipeline is optimised for reproducibility, automation and ability to parallelize across sores and chromosomes (not speed _per se_) - it largely scales with number of unique variants being matched/scored and to a lesser extent samples as significant time is spent matching variants. It usually runs in a couple of hours on a personal laptop but 16 GB might be too little RAM if you have a lot of variants genotyped.

**Reply by Than0sss** [reply]
Thanks a lot for the valuable information!
